foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/jpa/jpql/ExpressionBuilderVisitor.java

```diff
@@ -996,7 +996,26 @@ public void visit(ExtractExpression expression) {
         queryExpression = queryExpression.extract(expression.getDatePart());
 
         // Set the expression type
-        type[0] = Object.class;
+        if (expression.hasFrom()) {
```

Comment:
add `@SuppressWarnings" fall-through

Reply:
Done. Also fixed `expression.hasFrom()` check which was not the best option here.